packages/components/src/internal/components/search/actions.ts

```diff
@@ -145,3 +159,119 @@ export function getProcessedSearchHits(
+            success: data => {
+                resolve(data.rows.map(row => row['Name']));
+            },
+            failure: reason => {
+                console.error("Problem retrieving filtered sample types", reason);
+                reject("There was a problem retrieving the filtered sample types. " + resolveErrorMessage(reason));
+            }
+        });
+    });
+}
+
+const SAMPLE_FINDER_VIEW_NAME = "Sample Finder";
+
+export function removeFinderGridView(model: QueryModel): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+        if (!model.isLoading) {
+            Query.deleteQueryView({
+                schemaName: model.schemaQuery.schemaName,
+                queryName: model.schemaQuery.queryName,
+                viewName: SAMPLE_FINDER_VIEW_NAME,
+                revert: true,
+                success: () => {
+                    resolve(true);
+                },
+                failure: (error) => {
+                    console.error("There was a problem deleting the Sample Finder view.", error);
+                    reject(resolveErrorMessage(error));
+                }
+            });
+        }
+    });
+}
+
+export function saveFinderGridView(schemaQuery: SchemaQuery, columns: any): Promise<SchemaQuery> {
+    return new Promise((resolve, reject) => {
+        Query.saveQueryViews({
+            schemaName: schemaQuery.schemaName,
+            queryName: schemaQuery.queryName,
+            // Mark the view as hidden, so it doesn't show up in LKS and in the grid view menus
+            views: [{ name: SAMPLE_FINDER_VIEW_NAME, columns, hidden: true }],
+            success: () => {
+                resolve(schemaQuery);
+            },
+            failure: response => {
+                console.error(response);
+                reject(
+                    'There was a problem creating the view for the data grid. ' + resolveErrorMessage(response)
+                );
+            },
+        });
+    });
+}
+
+function getSampleFinderConfigId(finderId: string, suffix: string): string {
+    return finderId + "|" + suffix;
+}
+
+export function getSampleFinderQueryConfigs(user: User, cards: FilterProps[], finderId: string): Promise<{[key: string]: QueryConfig}> {
+    const omittedColumns = getOmittedSampleTypeColumns(user);
+    const baseFilters = [];
+    const requiredColumns = [...SAMPLE_STATUS_REQUIRED_COLUMNS];
+    cards.forEach(card => {
+        if (card.filterArray.length) {
+            card.filterArray.forEach(filter => {
+                requiredColumns.push(filter.getColumnName());
+            });
+            baseFilters.push(...card.filterArray);
+        } else {
+            const cardColumnName = getFilterCardColumnName(card.entityDataType, card.schemaQuery);
+
+            requiredColumns.push(cardColumnName);
+            baseFilters.push(Filter.create(cardColumnName + "/Name", null, Filter.Types.NONBLANK));
+        }
```

Comment:
it might make sense to split out this building of the `baseFilters` array into a separate helper function for jest testing

Reply:
Done